test/cpp/jit/test_type_parser.cpp

```diff
@@ -0,0 +1,22 @@
+#include "test/cpp/jit/test_base.h"
+
+namespace c10 {
+//std::string serializeType(const Type &t);
+TypePtr parseType(const std::string& pythonStr);
+}
+
+namespace torch {
+namespace jit {
+void testTypeParser() {
+  std::string int_ps("int");
+  auto int_tp = c10::parseType(int_ps);
+  std::string int_tps = int_tp->python_str();
+  ASSERT_EQ(int_ps, int_tps);
+
+  std::string tuple_ps("Tuple[str, Optional[float], Dict[str, List[Tensor]], int]");
```

Comment:
This should have more tests for error cases: empty string, unmatched brackets, weird whitespace. If you can crash this parser then it is possible to construct a model that can be loaded to a mobile device that would crash the application, which is a security risk.

Reply:
Tests for error cases are added.